src/main/java/uk/ac/ebi/eva/evaseqcol/datasource/NCBIAssemblySequencesDataSource.java

```diff
@@ -0,0 +1,107 @@
+        GzipCompress gzipCompress = new GzipCompress();
+
+        Optional<Path> downloadFilePath = downloadAssemblySequences(accession, ncbiBrowser);
+        if (!downloadFilePath.isPresent()) {
+            return Optional.empty();
+        }
+        logger.info("Assembly sequences _fna.gz file downloaded successfully in: " + downloadFilePath);
+        // Uncompress the .gz file
+        Optional<Path> compressedFilePath = gzipCompress.unzip(downloadFilePath.get().toString(), asmFileDownloadDir);
+        if (!compressedFilePath.isPresent()){
+            return Optional.empty();
+        }
+
+        AssemblySequencesEntity assemblySequencesEntity;
+        try (InputStream stream = new FileInputStream(compressedFilePath.get().toFile())){
+            NCBIAssemblySequencesReader reader = readerFactory.build(stream, accession);
+            assemblySequencesEntity = reader.getAssemblySequenceEntity();
+            logger.info("NCBI: Assembly sequences' _fna.gz file with accession " + accession + " has been parsed successfully" );
+        } finally {
+            try {
+                ncbiBrowser.disconnect();
+                Files.deleteIfExists(downloadFilePath.get());
+                Files.deleteIfExists(compressedFilePath.get()); // Deleting the fasta file
+            } catch (IOException e) {
+                logger.warn("Error while trying to disconnect - ncbiBrowser (assembly: " + accession + ")");
+            }
+        }
+        return Optional.of(assemblySequencesEntity);
+    }
+
+
+    /**
+     * Download the assembly fna/fasta file given the accession and save it to /tmp
+     * After this method is called, the file will be downloaded, and the path to this file
+     * on your local computer will be returned*/
+    @Retryable(value = Exception.class, maxAttempts = 5, backoff = @Backoff(delay = 2000, multiplier = 2))
+    public Optional<Path> downloadAssemblySequences(String accession, NCBIBrowser ncbiBrowser) throws IOException {
+        // The same directory as the report file
+        Optional<String> directory = ncbiBrowser.getGenomeReportDirectory(accession);
+
+        if (!directory.isPresent()) {
+            return Optional.empty();
+        }
+
+        logger.info("NCBI directory for assembly genomic.fna download: " + directory.get());
+        FTPFile ftpFile = ncbiBrowser.getAssemblySequencesFastaFile(directory.get());
+        String ftpFilePath = directory.get() + ftpFile.getName();
+        Path downloadFilePath = Paths.get(asmFileDownloadDir, ftpFile.getName());
+        boolean success = ncbiBrowser.downloadFTPFile(ftpFilePath, downloadFilePath, ftpFile.getSize());
+        if (success) {
+            logger.info("NCBI assembly genomic.fna downloaded successfully (" + ftpFile.getName() + ")");
+            return Optional.of(downloadFilePath);
+        } else {
+            logger.error("NCBI assembly genomic.fna could not be downloaded successfully(" + ftpFile.getName() + ")");
```

Comment:
`logger.error("NCBI assembly genomic.fna could not be downloaded successfully(" + ftpFile.getName() + ")");`